### ISSUE TITLE
demo.py blocks at line 49  

### ISSUE THREAD

**Issue by lombardata** [scaelles/DEXTR-PyTorch#35]
Hi,
Thank you for your work!
I am using PyTorch and I've installed the required libraries in a conda environment in Python 3.6, but I haven't succeeded to reproduce the results you have shown in the GIF.
If I run the model in a Jupyter notebook the problem seems to be that the matplotlib library has problems with interactions, but when I try to launch the demo.py from the terminal with the following command :
python demo.py
the problem still remains. The image is not shown and I cannot click on the points (so the code enters a loop at line 49 :         extreme_points_ori = np.array(plt.ginput(4, timeout=0)).astype(np.int)
).

The installed libraries in my environment are :

_libgcc_mutex             0.1                        main  
_openmp_mutex             4.5                       1_gnu  
anyio                     2.2.0            py36h06a4308_1  
argon2-cffi               20.1.0           py36h27cfd23_1  
async_generator           1.10             py36h28b3542_0  
attrs                     21.2.0             pyhd3eb1b0_0  
babel                     2.9.1              pyhd3eb1b0_0  
bleach                    4.0.0              pyhd3eb1b0_0  
brotlipy                  0.7.0           py36h27cfd23_1003  
bzip2                     1.0.8                h7b6447c_0  
ca-certificates           2021.9.30            h06a4308_1  
cairo                     1.16.0               hf32fb01_1  
certifi                   2020.12.5        py36h06a4308_0  
cffi                      1.14.6           py36h400218f_0  
charset-normalizer        2.0.4              pyhd3eb1b0_0  
cloudpickle               2.0.0              pyhd3eb1b0_0  
contextvars               2.4                        py_0  
cryptography              3.4.7            py36hd23ed53_0  
cudatoolkit               10.0.130                      0  
cycler                    0.10.0                   py36_0  
cytoolz                   0.11.0           py36h7b6447c_0  
dask-core                 1.1.4                    py36_1  
dataclasses               0.8                pyh4f3eec9_6  
dbus                      1.13.18              hb2f20db_0  
decorator                 5.1.0              pyhd3eb1b0_0  
defusedxml                0.7.1              pyhd3eb1b0_0  
entrypoints               0.3                      py36_0  
expat                     2.4.1                h2531618_2  
ffmpeg                    4.0                  hcdf2ecd_0  
fontconfig                2.13.1               h6c09931_0  
freeglut                  3.0.0                hf484d3e_5  
freetype                  2.10.4               h5ab3b9f_0  
glib                      2.69.1               h5202010_0  
graphite2                 1.3.14               h23475e2_0  
gst-plugins-base          1.14.0               h8213a91_2  
gstreamer                 1.14.0               h28cd5cc_2  
harfbuzz                  1.8.8                hffaf4a1_0  
hdf5                      1.10.2               hc401514_3    conda-forge
icu                       58.2                 he6710b0_3  
idna                      3.2                pyhd3eb1b0_0  
imageio                   2.9.0              pyhd3eb1b0_0  
immutables                0.16             py36h7f8727e_0  
importlib-metadata        4.8.1            py36h06a4308_0  
importlib_metadata        4.8.1                hd3eb1b0_0  
intel-openmp              2021.3.0          h06a4308_3350  
ipykernel                 5.3.4            py36h5ca1d4c_0  
ipympl                    0.7.0              pyhd3eb1b0_0  
ipython                   5.8.0                    py36_1    conda-forge
ipython_genutils          0.2.0              pyhd3eb1b0_1  
ipywidgets                7.6.5              pyhd3eb1b0_1  
jasper                    2.0.14               hd8c5072_2  
jinja2                    3.0.1              pyhd3eb1b0_0  
joblib                    1.0.1              pyhd3eb1b0_0  
jpeg                      9d                   h7f8727e_0  
json5                     0.9.6              pyhd3eb1b0_0  
jsonschema                3.2.0              pyhd3eb1b0_2  
jupyter_client            7.0.1              pyhd3eb1b0_0  
jupyter_core              4.8.1            py36h06a4308_0  
jupyter_server            1.4.1            py36h06a4308_0  
jupyterlab                3.1.7              pyhd3eb1b0_0  
jupyterlab_pygments       0.1.2                      py_0  
jupyterlab_server         2.8.2              pyhd3eb1b0_0  
jupyterlab_widgets        1.0.0              pyhd3eb1b0_1  
kiwisolver                1.3.1            py36h2531618_0  
lcms2                     2.12                 h3be6417_0  
ld_impl_linux-64          2.35.1               h7274673_9  
libblas                   3.9.0           11_linux64_openblas    conda-forge
libcblas                  3.9.0           11_linux64_openblas    conda-forge
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.3.0               h5101ec6_17  
libgfortran               3.0.0                         1    conda-forge
libgfortran-ng            9.3.0               ha5ec8a7_17  
libgfortran5              9.3.0               ha5ec8a7_17  
libglu                    9.0.0                hf484d3e_1  
libgomp                   9.3.0               h5101ec6_17  
liblapack                 3.9.0           11_linux64_openblas    conda-forge
libopenblas               0.3.17          pthreads_h8fe5266_1    conda-forge
libopencv                 3.4.2                hb342d67_1  
libopus                   1.3.1                h7b6447c_0  
libpng                    1.6.37               hbc83047_0  
libsodium                 1.0.18               h7b6447c_0  
libstdcxx-ng              9.3.0               hd4cf53a_17  
libtiff                   4.2.0                h85742a9_0  
libuuid                   1.0.3                h7f8727e_2  
libvpx                    1.7.0                h439df22_0  
libwebp-base              1.2.0                h27cfd23_0  
libxcb                    1.14                 h7b6447c_0  
libxml2                   2.9.12               h03d6c58_0  
lz4-c                     1.9.3                h295c915_1  
markupsafe                2.0.1            py36h27cfd23_0  
matplotlib                3.3.4            py36h06a4308_0  
matplotlib-base           3.3.4            py36h62a2d02_0  
mistune                   0.8.4            py36h7b6447c_0  
mkl                       2021.3.0           h06a4308_520  
nbclassic                 0.2.6              pyhd3eb1b0_0  
nbclient                  0.5.3              pyhd3eb1b0_0  
nbconvert                 6.0.7                    py36_0  
nbformat                  5.1.3              pyhd3eb1b0_0  
ncurses                   6.2                  he6710b0_1  
nest-asyncio              1.5.1              pyhd3eb1b0_0  
networkx                  2.2                      py36_1  
ninja                     1.10.2               hff7bd54_1  
notebook                  6.4.3            py36h06a4308_0  
numpy                     1.19.5           py36hfc0c790_2    conda-forge
olefile                   0.46                     py36_0  
opencv                    3.4.2            py36h6fd60c2_1  
openssl                   1.1.1l               h7f8727e_0  
packaging                 21.0               pyhd3eb1b0_0  
pandoc                    2.12                 h06a4308_0  
pandocfilters             1.4.3            py36h06a4308_1  
pcre                      8.45                 h295c915_0  
pexpect                   4.8.0              pyhd3eb1b0_3  
pickleshare               0.7.5           pyhd3eb1b0_1003  
pillow                    8.2.0            py36he98fc37_0  
pip                       21.0.1           py36h06a4308_0  
pixman                    0.40.0               h7f8727e_1  
prometheus_client         0.11.0             pyhd3eb1b0_0  
prompt_toolkit            1.0.15                     py_1    conda-forge
ptyprocess                0.7.0              pyhd3eb1b0_2  
py-opencv                 3.4.2            py36hb342d67_1  
pycparser                 2.20                       py_2  
pygments                  2.10.0             pyhd3eb1b0_0  
pyopenssl                 20.0.1             pyhd3eb1b0_1  
pyparsing                 2.4.7              pyhd3eb1b0_0  
pyqt                      5.9.2            py36h05f1152_2  
pyrsistent                0.17.3           py36h7b6447c_0  
pysocks                   1.7.1            py36h06a4308_0  
python                    3.6.13               h12debd9_1  
python-dateutil           2.8.2              pyhd3eb1b0_0  
python_abi                3.6                     2_cp36m    conda-forge
pytorch                   1.2.0           py3.6_cuda10.0.130_cudnn7.6.2_0    pytorch
pytz                      2021.3             pyhd3eb1b0_0  
pywavelets                1.1.1            py36h7b6447c_2  
pyzmq                     22.2.1           py36h295c915_1  
qt                        5.9.7                h5867ecd_1  
readline                  8.1                  h27cfd23_0  
requests                  2.26.0             pyhd3eb1b0_0  
scikit-image              0.17.2           py36h284efc9_4    conda-forge
scikit-learn              0.24.2           py36ha9443f7_0  
scipy                     1.5.3            py36h9e8f40b_0    conda-forge
send2trash                1.8.0              pyhd3eb1b0_1  
setuptools                58.0.4           py36h06a4308_0  
simplegeneric             0.8.1                    py36_2  
sip                       4.19.8           py36hf484d3e_0  
six                       1.16.0             pyhd3eb1b0_0  
sniffio                   1.2.0            py36h06a4308_1  
sqlite                    3.36.0               hc218d9a_0  
terminado                 0.9.4            py36h06a4308_0  
testpath                  0.5.0              pyhd3eb1b0_0  
threadpoolctl             2.2.0              pyh0d69192_0  
tifffile                  2020.10.1        py36hdd07704_2  
tk                        8.6.11               h1ccaba5_0  
toolz                     0.11.1             pyhd3eb1b0_0  
torchvision               0.4.0                py36_cu100    pytorch
tornado                   6.1              py36h27cfd23_0  
traitlets                 4.3.3            py36h06a4308_0  
typing-extensions         3.10.0.2             hd3eb1b0_0  
typing_extensions         3.10.0.2           pyh06a4308_0  
urllib3                   1.26.7             pyhd3eb1b0_0  
wcwidth                   0.2.5              pyhd3eb1b0_0  
webencodings              0.5.1                    py36_1  
wheel                     0.37.0             pyhd3eb1b0_1  
widgetsnbextension        3.5.1                    py36_0  
xz                        5.2.5                h7b6447c_0  
zeromq                    4.3.4                h2531618_0  
zipp                      3.6.0              pyhd3eb1b0_0  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.4.9                haebb681_0  

Can you please help me?
Bests,
Matteo
